src/core/phelia.ts

```diff
@@ -558,6 +601,76 @@ export class Phelia {
     const executedCallbacks = new Map<string, boolean>();
     const executionPromises = new Array<Promise<any>>();
 
+    /** Extracts content from modal form submission */
+    const formFromPayload = (payload: any) => {
+      if (payload.type === "view_submission") {
+        const form = Object.keys(payload.view.state.values)
+          .map((key) => [key, Object.keys(payload.view.state.values[key])[0]])
```

Comment:
That could be simplified, without the need for 2 `map` calls.

You can just build `action` on the fly inside the 2nd map, along with `data`

```javascript
const action = Object.keys(payload.view.state.values[key])[0]
```